Makefile

```diff
@@ -0,0 +1,167 @@
+TEMPLATE = $(MODERNCVDIR)/template.tex
+TEMPLATEBIB = publications.bib
+TEMPLATEBASE = $(basename $(TEMPLATE))
+TEMPLATEPDF = $(addsuffix .pdf,$(TEMPLATEBASE))
+MANUAL_BASE = $(basename $(MANUAL))
+MANUALPDF = $(addsuffix .pdf,$(MANUAL_BASE))
+
+# redefine the template rule depending on whether the user has specified STYLE
+# or not.
+ifdef STYLE
+	# in this case user has specified STYLE
+else
+	STYLE = casual
+endif
+template: $(TEMPLATE) $(TEMPLATEBIB)
+	if [[ $(strip $(STYLE)) == "casual" ]]; then
+		# build template in default style
+		latexmk -pdflua -bibtex -quiet "$<"
+	else
+		# build template in style $(STYLE). This assumes that casual is the default.
+		sedstring="s/moderncvstyle{casual}/moderncvstyle{$(STYLE)}/g"
+		sed -i $$sedstring $(TEMPLATE)
+		# build template in specified style
+		latexmk -pdflua -bibtex -quiet "$<"
+		# reset template to default value
+		sedstring="s/moderncvstyle{$(STYLE)}/moderncvstyle{casual}/g"
+		sed -i $$sedstring $(TEMPLATE)
+	fi
+
+
+templates: $(TEMPLATE) $(TEMPLATEBIB)
+	# build the template for each style and store pdfs in the examples folder
+	# this is done to include these expamples in release tar balls.
+	mkdir -p $(EXAMPLESDIR)
+	previousstyle="casual"
+	for style in casual classic banking oldstyle fancy; do
+		sedstring="s/moderncvstyle{$$previousstyle}/moderncvstyle{$$style}/g"
+		sed -i $$sedstring $(TEMPLATE)
+		latexmk -pdflua -bibtex -quiet $(TEMPLATE)
+		cp $(TEMPLATEPDF) $(EXAMPLESDIR)/$(TEMPLATEBASE)-$$style.pdf
+		mv $(TEMPLATEPDF) $(MANUALDIR)/$(TEMPLATEBASE)-$$style.pdf
+		previousstyle=$$style
+		unset sedstring
+	done
+	sedstring="s/moderncvstyle{$$previousstyle}/moderncvstyle{casual}/g"
+	sed -i $$sedstring $(TEMPLATE)
+
+
+userguide: templates $(MANUAL)
+	# build the user guide. Since the guide includes the template examples, we
+	# build those first by calling the templates rule.
+	cd $(MANUALDIR)
+	./format_files_for_userguide.py 
+	lualatex $(MANUALTEX)
+	lualatex $(MANUALTEX)
+	cd ..
+
+
+version:
+	# Upate version information and date of all moderncv files. A call make version
+	# will define VERSIONNEXT=$(shell git describe --tags). Alternatively, call
+	# "make version NEW=v5.13.298" to set v5.13.298 as the new version number.
+	# The date gets calculated by the date function.
+	if [[ $(strip $(VERSION)) == $(strip $(VERSIONNEXT)) ]]; then
+		echo "Old version number $(VERSION)  same as new version number $(VERSIONNEXT)"
+		echo "nothing to do, aborting."
+	else
+		# we need to split the $(VERSION) date format into substrings to using
+		# sed. This is due to / being a special character in sed.
+		YEAR=$(shell echo $(VERSIONDATE) | cut -d "/" -f 1)
+		MONTH=$(shell echo $(VERSIONDATE) | cut -d "/" -f 2)
+		DAY=$(shell echo $(VERSIONDATE) | cut -d "/" -f 3)
+		# update version info and date of *.sty, *.cls and *.tex files
+		# prepare find and replace with sed
+		findstr="$$YEAR\\/$$MONTH\\/$$DAY $(VERSION)"
+		replacestr="$(VERSIONDATENEXT) $(VERSIONNEXT)"
```

Comment:
Ok I see now why you want to have the date in the makefile, but I still think this is not a particular good design. You can use this regex `\d{4}\-\d{2}\-\d{2}\sv\d+.\d+.\d+` (via `sed -E`) to find the dates, and replace it with `$(date -u -I) $(NEWVERSION)`

Before, check if `NEWVERSION` is set, if not, then abort.